subiquity/server/controllers/install.py

```diff
@@ -298,6 +299,23 @@ async def install_oem_metapackages(ctx):
         if not self.model.oem.metapkgs:
             return
 
+        # cause existing kernels to be removed.  If a kernel is already in the
+        # minimal layer, it might not be the intended one, so tell kernel
+        # to remove that pre-installed kernel.  This can't use the curtin
+        # ensure_one_kernel logic as by the time it knows what to do, there are
+        # already two kernels, so instead we tell curtin explicitly what to do.
+        target = self.tpath()
+        if self.app.opts.dry_run:
+            existing_kernels = []
+        else:
+            existing_kernels = list_kernels(target=target)
+        for metapkg in self.model.oem.metapkgs:
+            if metapkg.name in existing_kernels:
```

Comment:
isn't metapkg.name going to be something like oem-somerville-weedle-meta, not something like linux-image-6.5.0-1026-oem ?

Reply:
correct

Reply:
per chat elsewhere, I do think this part is wrong and will revisit it